pennylane/templates/subroutines/qsvt.py

```diff
@@ -21,131 +21,203 @@
-        are matrix-based operators and well-suited for simulators.
-        To implement QSVT with user-defined circuits for the block encoding and
-        projector-controlled phase shifts, use the :class:`~.QSVT` template.
+        \begin{pmatrix}
+        A & * \\
+        * & * \\
+        \end{pmatrix}
+        \xrightarrow{QSVT}
+        \begin{pmatrix}
+        \text{poly}(A) + i \dots & * \\
+        * & * \\
+        \end{pmatrix}
 
-    Given a matrix :math:`A`, and a list of angles :math:`\vec{\phi}`, this function applies a
-    circuit for the quantum singular value transformation using :class:`~.BlockEncode` and
-    :class:`~.PCPhase`.
+    The polynomial transformation is encoded as the real part of the top left term after applying the operator.
 
-    When the number of angles is even (:math:`d` is odd), the QSVT circuit is defined as:
+    This function calculates the required phase angles from the polynomial using :func:`~.math.poly_to_angles`.
 
-    .. math::
+    .. note::
 
-        U_{QSVT} = \tilde{\Pi}_{\phi_1}U\left[\prod^{(d-1)/2}_{k=1}\Pi_{\phi_{2k}}U^\dagger
-        \tilde{\Pi}_{\phi_{2k+1}}U\right]\Pi_{\phi_{d+1}},
+        The function ``poly_to_angles``, used within ``qsvt``, is not JIT-compatible, which prevents ``poly`` from being
+        traceable in ``qsvt``. However, ``A`` is traceable and can be optimized by JIT within this function.
 
+    Args:
 
-    and when the number of angles is odd (:math:`d` is even):
+        A (Union[tensor_like, Operator]): The matrix on which the QSVT will be applied.
+            This can be an array or an object that has a Pauli representation.
 
-    .. math::
+        poly (tensor_like): Polynomial coefficients defining the transformation, represented in increasing order of degree.
+            This means the first coefficient corresponds to the constant term, the second to the linear term, and so on.
 
-        U_{QSVT} = \left[\prod^{d/2}_{k=1}\Pi_{\phi_{2k-1}}U^\dagger\tilde{\Pi}_{\phi_{2k}}U\right]
-        \Pi_{\phi_{d+1}}.
+        encoding_wires (Sequence[int]): The qubit wires used for the block encoding. See Usage Details below for
+            more information on ``encoding_wires`` depending on the block encoding used.
 
-    Here, :math:`U` denotes a block encoding of :math:`A` via :class:`~.BlockEncode` and
-    :math:`\Pi_\phi` denotes a projector-controlled phase shift with angle :math:`\phi`
-    via :class:`~.PCPhase`.
+        block_encoding (str): Specifies the type of block encoding to use. Options include:
 
-    This circuit applies a polynomial transformation (:math:`Poly^{SV}`) to the singular values of
-    the block encoded matrix:
+            - ``"prepselprep"``: Embeds the Hamiltonian ``A`` using :class:`~pennylane.PrepSelPrep`. Default encoding for Hamiltonians.
+            - ``"qubitization"``: Embeds the Hamiltonian ``A`` using :class:`~pennylane.Qubitization`.
```

Comment:
It is a fair point. In terms of efficiency it always makes more sense to use prepselprep. The reason I include it is:

- to emphasize the idea that qubitization is a block encoding (which is not super widespread).

- enable all the block encodings we have available in PennyLane so that the user has the power to do what he wants. Maybe you want to verify that QSVT gives the same results with one method and with another.

If you think it is still better to remove it, we can discuss it :)